zuul-core/src/main/java/com/netflix/zuul/netty/server/ClientConnectionsShutdown.java

```diff
@@ -113,7 +113,14 @@ Promise<Void> gracefullyShutdownClientChannels(boolean forceCloseAfterTimeout) {
             ScheduledFuture<?> timeoutTask = executor.schedule(
                     () -> {
                         LOG.warn("Force closing remaining {} active client channels.", channels.size());
-                        channels.close();
+                        channels.close().addListener(future -> {
```

Comment:
This shouldn't be necessary.

>The graceful close promise will not finish if some channels error out while closing. 

The `closeFuture` should be invoked even if certain channels error out on close(). It has the listener wired to set a success on the Promise. Where do you see this not being set?

Reply:
That's really odd. Anyway, this doesn't hurt 🚢